ehitchhiking-ui/src/app/shared/services/api.services/api.service.ts

```diff
@@ -33,22 +59,69 @@ export class ApiService {
 			}),
 			withCredentials: true,
 		};
-		return this.http.get(ApiService.apiUrl + URL_REGISTRY.CURRENT_USER, httpOptions);
+		return this.http
+			.get(ApiService.apiUrl + URL_REGISTRY.CURRENT_USER, httpOptions)
+			.catch((error) => {
+				if (error.status === 401) {
+					this.showResponseMessage(ResponseMessages.Error, 'Invalid login or password!');
+					return Promise.reject();
+				}
+				this.showResponseMessage(ResponseMessages.Error, 'Api Error Occurred!');
+				return Promise.reject();
+			})
+			.toPromise();
 	}
 
 	private generateRequest(
 		type: RequestMethods,
 		urlPath: string,
-		data: any,
-		isCacheable: boolean = false
-	): Observable<any> {
+		isCacheable: boolean = false,
+		isNotified: boolean = true,
+		data?: any
+	): Promise<any> {
 		const body = data;
 		let url = ApiService.apiUrl + urlPath;
+		let responseObservable = null;
 		if (type === RequestMethods.GET || type === RequestMethods.DEL) {
 			url = this.insertParameters(url, data);
-			return this.http.request(type, url, this.getRequestOptions(isCacheable) as any);
 		}
-		return this.http.request(type, url, this.getRequestOptions(isCacheable, body) as any);
+		responseObservable = this.http.request(type, url, this.getRequestOptions(isCacheable, body) as any);
+		return this.responseMapping(responseObservable, isNotified).toPromise();
+	}
+
+	private responseMapping(
+		newResponse: Observable<HttpResponse<any> | HttpErrorResponse>,
+		isNotified: boolean = true
+	): any {
+		return newResponse.pipe(
+			map((response: HttpResponse<any>) => {
+				const {msg, status, data} = response.body || response;
+				if (isNotified && ResponseMessages.Error === status) {
```

Comment:
if (ResponseMessages.Error === status) {
          if (isNotified) {
            this.showResponseMessage(ResponseMessages.Error, msg);  
          }
					
     return Promise.reject(msg);
}

if (ResponseMessages.Ok === status) {
          if (isNotified) {
            this.showResponseMessage(ResponseMessages.Ok, msg); 
          }
				return data;	
}

return data;

Reply:
done